spec/active_record_consume/batch_consumption_spec.rb

```diff
@@ -100,4 +99,24 @@
       end
     end
   end
+
+  describe '#consume_batch' do
+    describe 'post_process in compact mode' do
+      let(:payload) do
+        [
+          { v: 1 }, { v: 2 }, { v: 3 }, { v: 4 }, { v: 5 }
+        ]
+      end
+      let(:metadata) do
+        { keys: [1, 2, 3, 4, 5] }
+      end
+
+      it 'should be called once with a flat array of records' do
+        records = [Object, Object] # should be ActiveRecord object
+        expect(batch_consumer).to receive(:post_process).once.with(records)
+        expect(batch_consumer).to receive(:update_database) { records }
+        batch_consumer.send(:consume_batch, payload, metadata)
+      end
```

Comment:
I think we're probably missing some more tests - e.g. with and without max\_db\_batch\_size, and with and without deletions. Probably can just add another `expect` onto existing tests showing that they return the expected values.